odml/property.py

```diff
@@ -373,45 +373,73 @@ def clone(self):
+            new_value = self._convert_value_input(source.value)
+            if not self._validate_values(new_value):
+                raise ValueError("odml.Property.merge: passed value(s) cannot "
+                                 "be converted to data type '%s'!" % self._dtype)
+
+        if self.unit is not None and source.unit is not None and self.unit != source.unit:
+            raise ValueError("odml.Property.merge: "
+                             "src and dest units (%s, %s) do not match!" %
+                             (source.unit, self.unit))
+
+        if self.definition is not None and source.definition is not None:
             self_def = ''.join(map(str.strip, self.definition.split())).lower()
-            other_def = ''.join(map(str.strip, other.definition.split())).lower()
+            other_def = ''.join(map(str.strip, source.definition.split())).lower()
             if self_def != other_def:
-                raise ValueError("odml.Property.merge: src and dest definitions do not match!")
+                raise ValueError("odml.Property.merge: "
+                                 "src and dest definitions do not match!")
 
-        if self.uncertainty is not None and other.uncertainty is not None:
-            raise ValueError("odml.Property.merge: src and dest uncertainty both set and do not match!")
+        if (self.uncertainty is not None and source.uncertainty is not None and
+                self.uncertainty != source.uncertainty):
+            raise ValueError("odml.Property.merge: "
+                             "src and dest uncertainty both set and do not match!")
 
-        if self.reference is not None and other.reference is not None:
+        if self.reference is not None and source.reference is not None:
             self_ref = ''.join(map(str.strip, self.reference.lower().split()))
-            other_ref = ''.join(map(str.strip, other.reference.lower().split()))
+            other_ref = ''.join(map(str.strip, source.reference.lower().split()))
             if self_ref != other_ref:
-                raise ValueError("odml.Property.merge: src and dest references are in conflict!")
+                raise ValueError("odml.Property.merge: "
+                                 "src and dest references are in conflict!")
 
-        if self.value_origin is not None and other.value_origin is not None:
+        if self.value_origin is not None and source.value_origin is not None:
             self_ori = ''.join(map(str.strip, self.value_origin.lower().split()))
-            other_ori = ''.join(map(str.strip, other.value_origin.lower().split()))
+            other_ori = ''.join(map(str.strip, source.value_origin.lower().split()))
             if self_ori != other_ori:
-                raise ValueError("odml.Property.merge: src and dest value_origin are in conflict!")
+                raise ValueError("odml.Property.merge: "
+                                 "src and dest value_origin are in conflict!")
```

Comment:
yet another question: do we want to raise errors in the check function? Or, do we want to return a bool?

Reply:
Since there are various different Error messages narrowing down what went wrong I'd keep it that way.